Signal/src/environment/NotificationsManager.m

```diff
@@ -312,9 +312,12 @@ - (void)notifyUserForIncomingMessage:(TSIncomingMessage *)message
                             [NSString stringWithFormat:NSLocalizedString(@"APN_MESSAGE_IN_GROUP_DETAILED", nil),
                                       senderName,
                                       threadName,
-                                      messageDescription];
+                                      [DisplayableText filterNotificationText:messageDescription]];
```

Comment:
I don't understand why this is desirable.  The message description isn't the format string.  So formatting specifiers in it shouldn't be escaped.  If they are (as you have it here), won't they be rendered incorrectly?  E.g. won't "I'm with you 100%" be rendered as "I'm with you 100%%"?

Reply:
Nope, that's the point of this change.

from https://developer.apple.com/documentation/uikit/uilocalnotification/1616646-alertbody

> Printf style escape characters are stripped from the string prior to display; to include a percent symbol (%) in the message, use two percent symbols (%%).

Reply:
I've included some more details, and a link to the docs in the comment in DisplayableText

Reply:
Oh. Can you please add a comment here (and maybe below too) with that link & quote?

Reply:
Done!